tsnet/tsnet_test.go

```diff
@@ -510,7 +510,7 @@ func TestStartStopStartGetsSameIP(t *testing.T) {
 			Dir:        tmps1,
 			ControlURL: controlURL,
 			Hostname:   "s1",
-			Logf:       logger.TestLogger(t),
+			Logf:       tstest.WhileTestRunningLogger(t),
```

Comment:
Could we not also achieve the same thing by waiting for close rather than deferring it at the end?

Reply:
```
panic: Log in goroutine after TestStartStopStartGetsSameIP has completed:
 ... health(warnable=no-derp-connection): error: Tailscale could not connect to the relay server with ID '0'. Your Internet connection might be down, or the server might be temporarily unavailable.
goroutine 14667 [running]:
testing.(*common).logDepth(0xc000354d00, {0xc012ae0000, 0xc8}, 0x3)
        /snap/go/10730/src/testing/testing.go:1029 +0x4c5
testing.(*common).log(...)
        /snap/go/10730/src/testing/testing.go:1011
testing.(*common).Logf(0xc000354d00, {0xc012a74000?, 0x185c661?}, {0xc01020c020?, 0x1885c0a?, 0x1e?})
        /snap/go/10730/src/testing/testing.go:1062 +0x54
tailscale.com/tsnet.TestStartStopStartGetsSameIP.TestStartStopStartGetsSameIP.func1.TestLogger.func2({0x1885c0a, 0x1e}, {0xc01020c020, 0x2, 0x2})
        /types/logger/logger.go:394 +0x94      
tailscale.com/tsnet.(*Server).start.func1({0x1885c0a?, 0x94?}, {0xc01020c020?, 0x16?, 0x417845?})
        /tsnet/tsnet.go:517 +0x96
tailscale.com/ipn/ipnlocal.(*LocalBackend).onHealthChange(0xc00081c008, 0xc0002fe700, 0xc01545c000)
        /ipn/ipnlocal/local.go:845 +0x109      
created by tailscale.com/health.(*Tracker).setUnhealthyLocked.func1 in goroutine 14666
        /health/health.go:402 +0x13a
FAIL    tailscale.com/tsnet     13.126s
FAIL
```
The logging (and panic) occurs in a fire-and-forget goroutine:
https://github.com/tailscale/tailscale/blob/52ef27ab7c3078513ea7e265ce4a1b3e0adea833/health/health.go#L397-L404

Reply:
That's a regrettable choice.

Reply:
It used to be worse. :)